src/coreclr/jit/helperexpansion.cpp

```diff
@@ -874,7 +873,12 @@ bool Compiler::fgExpandHelperForBlock(BasicBlock** pBlock)
 
         for (GenTree* const tree : stmt->TreeList())
         {
-            if (!tree->IsHelperCall())
+            if (!tree->IsCall())
+            {
+                continue;
+            }
+
+            if (!tree->IsHelperCall() && ((tree->AsCall()->gtCallMoreFlags & GTF_CALL_M_SPECIAL_INTRINSIC) == 0))
             {
                 continue;
             }
```

Comment:
Why this check? Just for TP purposes? Does it actually make a difference?

Reply:
You're right - it's no longer needed